### PR TITLE
net/http: add Transport.RoundTrip enforcement hook

### PR DESCRIPTION
**Pull request by bradfitz**
net/http: add enforcement hook to Transport.RoundTrip, like our net ones

Updates https://github.com/tailscale/go/issues/55
Updates https://github.com/tailscale/corp/issues/12702

### CODE REVIEW

File: src/net/http/tailscale.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package http
+
+var roundTripEnforcer func(*Request) error
+
+func SetRoundTripEnforcer(f func(*Request) error) {
```
Comment (maisem) [inline src/net/http/tailscale.go:13]: docs?

File: src/net/http/tailscale.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	if f == nil {
+		panic("nil func")
+	}
+	if roundTripEnforcer != nil {
```
Comment (maisem) [inline src/net/http/tailscale.go:17]: It's useful to be able to set this back to nil after a test runs. I've run into that issue for the other hooks that we have.

**Review by maisem — APPROVED** [review]

File: src/net/http/tailscale.go (hunk 2 above)
Comment (bradfitz) [reply src/net/http/tailscale.go:17]: It's meant to be installed exactly once in init. If we want per-test control, we provide that ourselves in our installed hook.

The other two hooks are like this too.

File: src/net/http/tailscale.go (hunk 1 above)
Comment (bradfitz) [reply src/net/http/tailscale.go:13]: done.

**Pull request merged**